Docker deployment and documentation: various improvements (#5353)
skyportal/skyportal: doc/deploy.md

## Building the Docker images

The first step is to create the images to launch. For that, you can start by git cloning the repository and checking out to the desired version. For a production deployment, we recommend using the version tagged on a release (see [versioning](versioning)).

> After customizing your `docker.yaml` file (the equivalent of `config.yaml`, used as the config file when building a docker image), you may build the images using the following command:

Author: I don't think so! It's literally what it is. We copy it as config.yaml in the Dockerfile! I think that on the other hand if we don't explain that, folks won't necessarily understand that they need to edit this file when building the image and not config.yaml

> After customizing your `docker.yaml` file (the equivalent of `config.yaml`, used as the config file when building a docker image), you may build the images using the following command:

Reviewer: Sorry, thought you were talking of Docker file, now see you're talking about config used in Docker. I wonder how we can make this even clearer.

```
make docker-local
```

## Starting containers

Next, we deploy two containers: `web` (the SkyPortal application, which image you just built) and
`db` (the PostgreSQL server, from the official Docker image).
The database is stored and persisted in a local docker volume called `skyportal_dbdata` (see `docker volume ls`), which location is specified in the `docker-compose.yaml` file, as well as the database name, user, and password, other volume mounts, and the port mapping.

To start the containers, run:
```
docker-compose up -d
```

The application will not be accessible until the database has been initialized, which is explained in the next section.

## Initializing the database

Note that by default, the SkyPortal image will run the application in production (`make run_production`, see the `CMD` directive at the end of the Dockerfile).

> The key behavior to note when running production mode is that the application will not create any database tables automatically (to avoid messing with production data). This means that the very first time you spin up SkyPortal using containers, the `skyportal_dbdata` volume will only have an empty database. While the `skyportal-web-1` container will start, the web application will not be accessible without an initialized database. This can easily be done manually, and the easiest way to do this is to run:

Author: Not sure what is the command for just initialize and empty DB, but I'll check

> The key behavior to note when running production mode is that the application will not create any database tables automatically (to avoid messing with production data). This means that the very first time you spin up SkyPortal using containers, the `skyportal_dbdata` volume will only have an empty database. While the `skyportal-web-1` container will start, the web application will not be accessible without an initialized database. This can easily be done manually, and the easiest way to do this is to run:

Reviewer: Won't a lot of people need that, if they want to start fresh?

> The key behavior to note when running production mode is that the application will not create any database tables automatically (to avoid messing with production data). This means that the very first time you spin up SkyPortal using containers, the `skyportal_dbdata` volume will only have an empty database. While the `skyportal-web-1` container will start, the web application will not be accessible without an initialized database. This can easily be done manually, and the easiest way to do this is to run:

Author: They would! I added a make command to create the tables.

Now, you may browse to SkyPortal at `http://localhost:5000` (or any other port you have configured it to run on).

## Customizing the deployment configuration

If you want to use a specific configuration file at runtime (that can then be different from the one used to build the image), you can mount it as a volume when starting the container. For example, if you had a `config.yaml` file in a directory called `config`, you would add to your `docker-compose.yaml` file:

```yaml
    web:
        ...
        volumes:
        - ./config/config.yaml:/etc/skyportal/config.yaml
    ```
```
This will mount the `config.yaml` file from the `config` directory to the `/etc/skyportal/config.yaml` path in the container. You could technically directly have it mounted to `/skyportal/config.yaml`, but we recommend using a different path to not lose track of the original configuration file used to build the image.

> You would need to pass the `--config=config.yaml` flag when calling any `make` commands. For the container to use this configuration file on startup, you can simply specify your own FLAG environment variable in the `docker-compose.yaml` file:

Reviewer: ```suggestion
You would need to pass the `--config=config.yaml` flag when calling any `make` commands. For the container to use this configuration file on startup, set the FLAG environment variable in the `docker-compose.yaml` file:
```
